### PR TITLE
Linear regression Example - custom numpy framework adapter 

### LINKED ISSUE #267: Simple ML model on NumPy as example of custom plugin in OpenFL
It would be great to integrate example of ML model training in pure NumPy as a showcase of custom plugin  implementation in OpenFL

### PR DESCRIPTION
**Pull request by igor-davidyuk**
closes #267

### CODE REVIEW

**Comment by igor-davidyuk** [comment]
I have read the CLA Document and I hereby sign the CLA

File: openfl-tutorials/interactive_api/linear_regression/workspace/SingleNotebook.ipynb [hunk 1]
```diff
@@ -0,0 +1,770 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4d6ae745-afa8-4bbe-ba4c-7346cd63d4cd",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!git clone https://github.com/igor-davidyuk/openfl.git -b linear-regression-numpy-tutorial\n",
```
Comment (itrushkin) [inline openfl-tutorials/interactive_api/linear_regression/workspace/SingleNotebook.ipynb]: This should be changed to original repository after merge.

File: openfl-tutorials/interactive_api/linear_regression/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# Linear Regression with Numpy and OpenFL
+1. Use `start_federation.ipynb` notebook to run Director and Envoys
+2. Follow `LinReg.ipynb` to train a simple model.
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/linear_regression/README.md]: Missing launch information in colab (haw to launch and preparation steps)

File: openfl-tutorials/interactive_api/linear_regression/workspace/custom_adapter.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+from openfl.plugins.frameworks_adapters.framework_adapter_interface import (
+    FrameworkAdapterPluginInterface,
+)
+
+
+class CustomFrameworkAdapter(FrameworkAdapterPluginInterface):
+    """Framework adapter plugin class."""
+
+    def __init__(self) -> None:
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/linear_regression/workspace/custom_adapter.py]: It's redundant, the same as in the parent class

File: openfl/component/envoy/envoy.py [hunk 4]
```diff
@@ -68,7 +68,9 @@ def run(self):
             self.is_experiment_running = True
             try:
                 with ExperimentWorkspace(
-                        experiment_name, data_file_path, is_install_requirements=True
+                        self.name + ' ' + experiment_name,
```
Comment (aleksandr-mokrov) [inline openfl/component/envoy/envoy.py:71]: Looks like the problem is not here, but in that you launch all envoys in one directory. By design they should have their own working directory, with their own config file, shard descriptor.

File: openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb [hunk 5]
```diff
@@ -0,0 +1,485 @@
...
+    }
+   },
+   "cell_type": "markdown",
+   "id": "c4b334ef-6a72-4b82-b978-1401973d0512",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "# We will use MSE as loss function and Ridge weights regularization\n",
+    "![image.png](attachment:99c798ac-85a8-48d6-bb31-a76508d20063.png)"
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb]: This image is not displayed in colab. And, maybe we will not store the picture in the git?

File: openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb [hunk 6]
```diff
@@ -0,0 +1,485 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "af89e7e5-6cfc-46bc-acd2-7d5bfb373091",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "#Define input array with angles from 60deg to 300deg converted to radians\n",
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb]: #Define -> # Define

File: openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb [hunk 7]
```diff
@@ -0,0 +1,485 @@
...
+    "x = np.array([i*np.pi/180 for i in range(60,300,4)])\n",
+    "np.random.seed(10)  #Setting seed for reproducibility\n",
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb]: #Setting -> # Setting

File: openfl-tutorials/interactive_api/linear_regression/workspace/SingleNotebook.ipynb [hunk 8]
```diff
@@ -0,0 +1,772 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dc9da235-02a8-4e7a-9455-5fe2462aa317",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "### Tasks\n",
+    "Using an Optimizer does not make sense for this experiment. Yet it is a required part of a trainingtask contract in current version of OpenFL, so we just pass None\n",
```
Comment (aleksandr-mokrov) [inline openfl-tutorials/interactive_api/linear_regression/workspace/SingleNotebook.ipynb]: `trainingtask` -> `training task`, `None` -> `None.`

**Review by aleksandr-mokrov — CHANGES_REQUESTED** [review]

File: openfl-tutorials/interactive_api/linear_regression/workspace/custom_adapter.py (hunk 3 above)
Comment (igor-davidyuk) [reply openfl-tutorials/interactive_api/linear_regression/workspace/custom_adapter.py]: removed, thanks

File: openfl/component/envoy/envoy.py (hunk 4 above)
Comment (igor-davidyuk) [reply openfl/component/envoy/envoy.py:71]: All our examples run envoys from a single workspace. I would say that it is a free feature.

File: openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb (hunk 5 above)
Comment (igor-davidyuk) [reply openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb]: replaced the image with a link

File: openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb (hunk 6 above)
Comment (igor-davidyuk) [reply openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb]: fixed. thanks

File: openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb (hunk 7 above)
Comment (igor-davidyuk) [reply openfl-tutorials/interactive_api/linear_regression/workspace/LinReg.ipynb]: fixed

File: openfl-tutorials/interactive_api/linear_regression/workspace/SingleNotebook.ipynb (hunk 8 above)
Comment (igor-davidyuk) [reply openfl-tutorials/interactive_api/linear_regression/workspace/SingleNotebook.ipynb]:
```suggestion
    "Using an Optimizer does not make sense for this experiment. Yet it is a required part of a training task contract in the current version of OpenFL, so we just pass None.\n",
```

File: openfl-tutorials/interactive_api/linear_regression/README.md [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+1. Start a Federation
+Distributed experiments:
+Use OpenFL CLI to start the Director and Envoy services from corresponding folders. 
+Single-node experiments:
+Users may use the same path or benefit from the start_federation notebook in the workspace folder
+
+2. Submit an experiment
+Follow LinReg jupyter notebook.
+
+### Google Colab
```
Comment (igor-davidyuk) [inline openfl-tutorials/interactive_api/linear_regression/README.md]:
```suggestion
### Google Colab:
```

File: openfl-tutorials/interactive_api/linear_regression/README.md (hunk 2 above)
Comment (igor-davidyuk) [reply openfl-tutorials/interactive_api/linear_regression/README.md]: Extended readme

**Review by aleksandr-mokrov — APPROVED** [review]

**Review by itrushkin — APPROVED** [review]

**Review by dmitryagapov — APPROVED** [review]

**Pull request merged**